pkg/ebpf/events_derived.go

```diff
@@ -116,3 +122,68 @@ func deriveContainerRemoved(t *Tracee) deriveFn {
 		return trace.Event{}, false, nil
 	}
 }
+
+func deriveDetectHookedSyscall(t *Tracee) deriveFn {
+	return func(event trace.Event) (trace.Event, bool, error) {
+		syscallsAdresses, err := getEventArgUlongArrVal(&event, "syscalls_addresses")
+		if err != nil {
+			return trace.Event{}, false, fmt.Errorf("error parsing syscalls_numbers arg: %v", err)
+		}
+		hookedSyscallData := analyzeHookedAddresses(syscallsAdresses, t.config.OsConfig, t.kernelSymbols)
+		de := event
+		de.EventID = int(DetectHookedSyscallsEventID)
+		de.EventName = "hooked_syscalls"
+		de.ReturnValue = 0
+		de.Args = []trace.Argument{
+			{ArgMeta: trace.ArgMeta{Name: "hooked_syscalls", Type: "hookedSyscallData[]"}, Value: hookedSyscallData},
+		}
+		de.ArgsNum = 1
+		return de, true, nil
+	}
+}
+
+func analyzeHookedAddresses(addresses []uint64, OsConfig *helpers.OSInfo, kernelSymbols *helpers.KernelSymbolTable) []bufferdecoder.HookedSyscallData {
+	hookedSyscallData := make([]bufferdecoder.HookedSyscallData, 0, 0)
+	arch := OsConfig.GetOSReleaseFieldValue(helpers.OS_ARCH)
+	var syscallsToCheck []int
+	if strings.Compare(arch, "x86_64") == 0 {
+		syscallsToCheck = syscallsToCheckX86
+	} else {
+		syscallsToCheck = syscallsToCheckArm
+	}
+	for idx, syscallsAdress := range addresses {
+		InTextSegment, err := kernelSymbols.TextSegmentContains(syscallsAdress)
+		if err != nil {
+			continue
+		}
+		if !InTextSegment {
+			hookingFunction := parseSymbol(syscallsAdress, kernelSymbols)
+			var syscallNumber int32
+			if idx > len(syscallsToCheck) {
+				continue
+			}
```

Comment:
Isn't it a bug if that happens?
If so, we should emit an error and break instead